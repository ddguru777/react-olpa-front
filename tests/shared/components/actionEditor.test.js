/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import ActionEditor from "shared/components/actionEditor";
import ActionsEditable from "shared/components/actionsEditable";

describe("components/actionEditor", () => {
  it("renders correctly", () => {
    const content = "some content";
    const component = renderer.create(<ActionEditor content={content} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders entity and variable with special characters", () => {
    const content = [
      `{{${encodeURIComponent("{{text}}")}}}`,
      `<<${encodeURIComponent("<<var>>")}>>`,
    ].join("");

    const wrapper = shallow(<ActionsEditable content={content} />);
    expect(wrapper.state("items")).toHaveLength(2);
  });
});
