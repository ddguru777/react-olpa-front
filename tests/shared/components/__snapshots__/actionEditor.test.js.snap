// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/actionEditor renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f5f5f5",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#eee",
        "display": "table",
        "marginBottom": "16px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "borderRight": "1px solid #ddd",
          "display": "table-cell",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <i
          className="material-icons"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          text_fields
        </i>
      </div>
      <div
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <i
          className="material-icons"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          all_out
        </i>
      </div>
      <div
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <i
          className="material-icons"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          assignment
        </i>
      </div>
      <span>
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <i
            className="material-icons"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex="0"
          >
            code
          </i>
        </div>
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <i
            className="material-icons"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex="0"
          >
            keyboard_return
          </i>
        </div>
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <i
            className="material-icons"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex="0"
          >
            insert_link
          </i>
        </div>
      </span>
    </div>
    <div
      style={
        Object {
          "borderRight": "1px solid #ddd",
          "display": "table-cell",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <i
          className="material-icons"
          disabled={true}
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          delete
        </i>
      </div>
    </div>
  </div>
  <div
    aria-label={null}
    className="contenteditable"
    id="ae_content"
    onBlur={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    spellCheck={false}
    tabIndex={0}
  >
    <span>
      <span
        className={undefined}
        data="text"
        id="ae_0"
        style={
          Object {
            "display": "inline-block",
            "padding": "0 4px",
          }
        }
        tabIndex={1}
      >
        some content
      </span>
    </span>
  </div>
</div>
`;
